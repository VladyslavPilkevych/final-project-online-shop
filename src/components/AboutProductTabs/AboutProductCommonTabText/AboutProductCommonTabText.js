import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';

import styles from './AboutProductCommonTabText.module.scss';

function AboutProductCommonTabText(props) {
  const {
    product: {
      name, model, capacity, storage, color, display, description, itemNo,
    },
  } = props;
  // console.log(name, model, color);

  const allProducts = useSelector((state) => state.products.products);
  // eslint-disable-next-line max-len
  const analogProducts = allProducts.filter((product) => product.name === name && product.model === model);

  const colors = [
    {
      name: 'shale black',
      hex: 'rgb(0, 0, 0)',
    },
    {
      name: 'gold',
      rgb: 'rgb(247, 232, 33)',
    },
    {
      name: 'silver',
      rgb: 'rgb(230, 235, 234)',
    },
    {
      name: 'space gray',
      rgb: 'rgb(185, 185, 185)',
    },
  ];

  const currentColor = colors.filter((colorObj) => colorObj.name === color);
  const colorsOfAnalogProducts = analogProducts.map((product) => product.color);
  console.log('colorsOfAnalogProducts', colorsOfAnalogProducts);
  return (
    <div>
      <h1 className={styles.productTitle}>
        {name}
        {model}
      </h1>
      <p className={styles.productDescription}>{description}</p>
      <div className={styles.colorsWrapper}>
        {colorsOfAnalogProducts.map((product, index) => (
          // eslint-disable-next-line max-len
          <div className={styles.colorItem} style={{ backgroundColor: product.color }} key={product.color} />

        ))}
      </div>
      <p className={styles.itemNo}>
        Article:
        {itemNo}
      </p>
      <p className={styles.contactUsText}>
        Have a Question?
        <a href="/contact-us">Contact us</a>
      </p>
    </div>
  );
}

AboutProductCommonTabText.propTypes = {
  product: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  name: PropTypes.string,
  model: PropTypes.string,
  capacity: PropTypes.string,
  storage: PropTypes.string,
  color: PropTypes.string,
  display: PropTypes.string,
  description: PropTypes.string,
  itemNo: PropTypes.string,
};

AboutProductCommonTabText.defaultProps = {
  product: {},
  name: '',
  model: '',
  capacity: '',
  storage: '',
  color: '',
  display: '',
  description: '',
  itemNo: '',
};

export default AboutProductCommonTabText;
