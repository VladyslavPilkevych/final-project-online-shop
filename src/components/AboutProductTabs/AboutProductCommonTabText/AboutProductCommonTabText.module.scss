// @import '../../../assets/styles/global.module.scss';

// .productTitle {
//     margin-top: 83px;
//     color: $black;
//     font-family: $fonts;
//     font-style: normal;
//     font-weight: 500;
//     font-size: 32px;
//     line-height: 48px;
// }

// .productDescription {
//     margin-top: 55px;
//     font-family: $fonts;
//     font-style: normal;
//     font-weight: 300;
//     font-size: 18px;
//     line-height: 30px;
// }

// .colorsWrapper {
//     margin-top: 30px;
// display: flex;
// }

// .colorItem {
//     border: 1px solid $black;
//     height: 30px;
//     width: 30px;
//     margin-right: 20px;
//     border-radius: 50%;
// }

// .itemNo {
//     text-align: right;
//     margin-top: 20px;
// }

// .contactUsText {
//     font-family: $fonts;
//     font-style: normal;
//     font-weight: 600;
//     font-size: 12px;
//     line-height: 18px;
//     padding-bottom: 100px;
//     margin-top: 20px;
// }