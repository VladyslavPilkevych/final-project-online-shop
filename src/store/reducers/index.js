import { combineReducers } from 'redux';
import preloaderReducer from './preloaderReducer';
import productsReducer from './productsReducer';
import menuReducer from './menuReducer';
import userReducer from './userReducer';
import searchReducer from './searchReducer';
import cartReducer from './cartReducer';
import miniMenuReducer from './miniMenuReducer';
import filtersCategoriesReducer from './filtersCategoriesReducer';

const reducer = combineReducers({
  loader: preloaderReducer,
  products: productsReducer,
  menu: menuReducer,
  miniMenu: miniMenuReducer,
  filtersCategories: filtersCategoriesReducer,
  user: userReducer,
  search: searchReducer,
  cart: cartReducer,
});

export default reducer;
