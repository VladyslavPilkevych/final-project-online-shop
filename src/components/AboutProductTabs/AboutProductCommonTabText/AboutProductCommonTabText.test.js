import { render } from '@testing-library/react';
import AboutProductCommonTabText from './AboutProductCommonTabText';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../../store';

function Component() {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <AboutProductCommonTabText />
      </Provider>
    </BrowserRouter>
  );
}

describe('Snapshot AboutProductCommonTabText should render', () => {
  test('Snapshot AboutProductCommonTabText is rendered', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});
