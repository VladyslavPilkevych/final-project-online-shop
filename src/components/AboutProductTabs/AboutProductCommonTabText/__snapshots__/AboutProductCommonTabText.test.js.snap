// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot AboutProductCommonTabText should render Snapshot AboutProductCommonTabText is rendered 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="productTitle"
    >
       
    </h1>
    <p
      class="productDescription"
    />
    <div
      class="colorsWrapper"
    />
    <p
      class="itemNo"
    >
      Article:
    </p>
    <p
      class="contactUsText"
    >
      Have a Question? 
      <a
        href="/contact-us"
      >
        Contact us
      </a>
    </p>
  </div>
</DocumentFragment>
`;
